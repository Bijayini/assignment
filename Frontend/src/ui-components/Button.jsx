import React from 'react';
import PropTypes from 'prop-types';


const Button = ({ children, type, size, variant, onClick, classList }) => {
  const getClassname = () =>
    `button ${size} ${variant} ${classList}`;

  return (
      <button type={type} onClick={onClick} className={getClassname()}>
        {children}
      </button>
  );
};

Button.defaultProps = {
  type: 'button',
  variant: 'primary',
  classList:'',
};

Button.propTypes = {
  variant: PropTypes.string,
  classList: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string,
  size: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default Button;
