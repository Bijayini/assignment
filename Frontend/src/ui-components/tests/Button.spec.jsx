import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';

describe('<Button />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      variant: 'secondary',
      size: 'small',
      onClick: jest.fn(),
    };
    component = shallow(<Button {...props}> Hello </Button>);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});