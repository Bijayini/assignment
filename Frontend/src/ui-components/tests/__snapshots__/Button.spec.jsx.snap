// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match with snapshot 1`] = `
"<button type=\\"button\\" onClick={[Function: mockConstructor]} className=\\"button small secondary\\">
   Hello 
</button>"
`;
